src/main/java/net/sf/jabref/gui/desktop/os/OSX.java

```diff
@@ -1,48 +1,47 @@
 package net.sf.jabref.gui.desktop.os;
 
 import net.sf.jabref.external.ExternalFileType;
-import net.sf.jabref.external.ExternalFileTypes;
-
 import java.awt.*;
 import java.io.File;
 import java.io.IOException;
+import java.nio.file.Paths;
 import java.util.Optional;
 
-public class OSX implements NativeDesktop {
+public class OSX extends NativeDesktop {
+
     @Override
-    public void openFile(String filePath, String fileType) throws IOException {
-        Optional<ExternalFileType> type = ExternalFileTypes.getInstance().getExternalFileTypeByExt(fileType);
-        if (type.isPresent() && !type.get().getOpenWithApplication().isEmpty()) {
-            openFileWithApplication(filePath, type.get().getOpenWithApplication());
+
+    public void openFile(String filePath, Optional<ExternalFileType> fileType) throws IOException {
+
+        if (hasRegisteredApplication(fileType)) {
+            openFileWithApplication(filePath, fileType.get().getOpenWithApplication());
         } else {
-            String[] cmd = { "/usr/bin/open", filePath };
+            String[] cmd = {"/usr/bin/open", filePath};
             Runtime.getRuntime().exec(cmd);
         }
     }
 
     @Override
-    public void openFileWithApplication(String filePath, String application) throws IOException {
-        // Use "-a <application>" if the app is specified, and just "open <filename>" otherwise:
-        String[] cmd = (application != null) && !application.isEmpty() ?
-                new String[] {"/usr/bin/open", "-a", application, filePath} :
-                new String[] {"/usr/bin/open", filePath};
+    protected void openFileWithApplication(String filePath, String application) throws IOException {
+        String[] cmd = {"/usr/bin/open", "-a", application, filePath};
```

Comment:
application is never null. a) it is only called when it has a registered application and b) getOpenWithApplication returns an empty String if nothing is found